cmd/thanos/receive.go

```diff
@@ -354,10 +354,14 @@ func runReceive(
 			return errors.Wrap(err, "setup gRPC server")
 		}
 
+		if conf.lazyRetrievalMaxBufferedResponses <= 0 {
```

Comment:
Maybe this validation can be moved into the `store.NewProxyStore` to keep this logic in one place? But if it was converted into a const then we wouldn't need this at all.